include/kalmar_math.h

```diff
@@ -26,17 +26,17 @@ extern "C" _Float16 __ocml_asinh_f16(_Float16 x) [[hc]];
 extern "C" float __ocml_asinh_f32(float x) [[hc]];
 extern "C" double __ocml_asinh_f64(double x) [[hc]];
 
-extern "C" _Float16 __ocml_atan_f16(_Float16 x) [[hc]];
-extern "C" float __ocml_atan_f32(float x) [[hc]];
-extern "C" double __ocml_atan_f64(double x) [[hc]];
+extern "C" _Float16 __ocml_atan_f16(_Float16 x) __attribute__((const, hc));
```

Comment:
Mostly because I was chasing a bug which was / is centred on `atan`, and the attributes we plop on the forward declare do not match the ones ROCDL actually uses. There are probably quite a few other mismatches lingering, and to be fair this tends to sort of work / was a red-herring in the context of the bug I was chasing, so I think I'll just skip this bit for this commit.